plugin/src/lib.rs

```diff
@@ -319,8 +319,8 @@ impl<'a, 'cx> Folder for MutatorPlugin<'a, 'cx> {
                             ],
                         );
                     }
-                    let left = left.map(|e| fold::noop_fold_expr(e, self));
-                    let right = right.map(|e| fold::noop_fold_expr(e, self));
+                    let left = left.map(|e| self.fold_expr(P(e)));
```

Comment:
if you run `cargo +nightly test` on the "plugin" folder you will see that this line is not compiling. If I'm not wrong, you need to change it to:

```rust
let left = self.fold_expr(left);
```
and the same with the other changes.

Note that `fold::noop_fold_expr` expects `Expr`, and because of this, the `map` was needed. On the other hand, `self.fold_expr` expects a `P(Expr)` and returns `P(Expr)`, so the map is not needed.